### DISCUSSION TITLE
Blog comment on How to run an embedded vector database in 10 lines of code

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#1045] [Docs/blog comments]
# Blog comment on How to run an embedded vector database in 10 lines of code

The Weaviate server can be run locally directly from client code

https://weaviate.io/blog/embedded-local-weaviate

**Comment by tdubon** [comment]
This was an awesome tutorial, thanks! It does a great job bridging the gap between using a sandbox example to using locally stored data.

**Comment by nobutoba** [comment]
This is just a friendly reminder that a closing parenthesis is missing in the CI/CD Python code:
```python
client = weaviate.Client(
    embedded_options=weaviate.embedded.EmbeddedOptions(),
    additional_headers={
        'X-OpenAI-Api-Key': 'YOUR-OPENAI-API-KEY'  # Replace w/ your OPENAI API key
    }
) <== missing in the sample code
```

**Reply by databyjp** [reply]
Well spotted, thank you :)

**Comment by Tej-Sharma** [comment]
Could you please add a searching example? I keep getting a 'TypeError: fetch failed' even though the adding nodes was working
